### PR TITLE
Added support for GitHub Enterprise to github_deploy_key module

### PR DESCRIPTION
**Pull request by bincyber**
##### SUMMARY

Updated github_deploy_key to support GitHub Enterprise by allowing users to specify the URL for the GitHub API

Feature requested in #59560

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - `github_deploy_key`

##### ANSIBLE VERSION

```
ansible 2.8.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/vagrant/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.7.3 (default, May 11 2019, 00:45:16) [GCC 8.3.1 20190223 (Red Hat 8.3.1-2)]
```

### CODE REVIEW

**Comment by bincyber** [comment]
The above issues will be fixed when #36876 is merged and this commit is rebased to include those changes.

File: lib/ansible/modules/source_control/github_deploy_key.py [hunk 1]
```diff
@@ -241,6 +246,7 @@ def main():
 
     module = AnsibleModule(
         argument_spec=dict(
+            github_url=key=dict(required=False, type='str', default="https://api.github.com"),
```
Comment (Akasurde) [inline lib/ansible/modules/source_control/github_deploy_key.py]:
```suggestion
            github_url=dict(default="https://api.github.com"),
```

Comment (Akasurde) [reply lib/ansible/modules/source_control/github_deploy_key.py]: `key=` is not required. `required=False` and `type='str'` are defaults so no need to mention.

File: lib/ansible/modules/source_control/github_deploy_key.py [hunk 2]
```diff
@@ -21,6 +21,11 @@
   - "Adds or removes deploy keys for GitHub repositories. Supports authentication using username and password,
   username and password and 2-factor authentication code (OTP), OAuth2 token, or personal access token."
 options:
+  github_url:
+    description:
+      - The base URL of the GitHub API
+    required: false
```
Comment (Akasurde) [inline lib/ansible/modules/source_control/github_deploy_key.py]:
```suggestion
    required: false
    type: str
    version_added: "2.9"
```

File: lib/ansible/modules/source_control/github_deploy_key.py (hunk 1 above)
Comment (bincyber) [reply lib/ansible/modules/source_control/github_deploy_key.py]: Fixed

File: lib/ansible/modules/source_control/github_deploy_key.py (hunk 2 above)
Comment (bincyber) [reply lib/ansible/modules/source_control/github_deploy_key.py]: This module was added in 2.4. This PR is just an update.

Comment (bincyber) [reply lib/ansible/modules/source_control/github_deploy_key.py]: Fixed

**Comment by bincyber** [comment]
ready_for_review

**Comment by bincyber** [comment]
@Akasurde @gundalow please review

**Comment by gundalow** [comment]
Thanks!

**Pull request merged**